src/all/unionmangas/src/eu/kanade/tachiyomi/extension/all/unionmangas/UnionMangas.kt

```
@@ -0,0 +1,231 @@
+            .apply { timeZone = TimeZone.getTimeZone("GMT") }
+            .format(Date())
+        val path = url.toUrlWithoutDomain()
+
+        return headersBuilder()
+            .add("_hash", authorization(apiSeed, domain, date))
+            .add("_tranId", authorization(apiSeed, domain, date, path))
+            .add("_date", date)
+            .add("_domain", domain)
+            .add("_path", path)
+            .add("Origin", baseUrl)
+            .add("Host", apiUrl.removeProtocol())
+            .add("Referer", "$baseUrl/")
+            .build()
+    }
+
+    private fun authorization(vararg payloads: String): String {
+        val md = MessageDigest.getInstance("MD5")
+        val bytes = payloads.joinToString("").toByteArray()
+        val digest = md.digest(bytes)
+        return digest
+            .fold("") { str, byte -> str + "%02x".format(byte) }
+            .padStart(32, '0')
+    }
+
+    override fun fetchChapterList(manga: SManga): Observable<List<SChapter>> {
+        val chapters = mutableListOf<SChapter>()
+        var currentPage = 0
+        do {
+            val chaptersDto = fetchChapterListPageable(manga, currentPage)
+            chapters += chaptersDto.toModel(langOption)
+            currentPage++
+        } while (chaptersDto.hasNextPage())
+        return Observable.from(listOf(chapters.reversed()))
+    }
+
+    private fun fetchChapterListPageable(manga: SManga, page: Int): ChapterPageDto {
+        return try {
+            val maxResult = 16
+            val pathSearch = when (langOption.lang) {
+                "it" -> langOption.infix
+                else -> "v3/po"
+            }
+            val url = "$apiUrl/api/$pathSearch/GetChapterListFilter/${manga.slug()}/$maxResult/$page/all/ASC"
+            return client
+                .newCall(GET(url, apiHeaders(url)))
+                .execute()
+                .toChapterPageDto()
+        } catch (e: Exception) {
```

don't catch too generic of a exception, not sure if you need to catch any exceptions since any failed call means missing chapters

---

For any exception, return a predefined DTO and then the condition that the next page exists will be false, stopping the chapter retrieval cycle.

Or would it be better to report the error via Toast?